xbmc/addons/AddonDll.h

```diff
@@ -46,6 +46,28 @@ namespace ADDON
 
     bool DllLoaded(void) const;
 
+    /*!
+     * @brief Function to create a addon instance class
+     *
+     * @param[in] instanceType The wanted instance type class to open on addon
+     * @param[in] instanceID The from Kodi used ID string of active instance
+     * @param[in] instance Pointer where the interface functions from addon
+     *                     becomes stored during his instance creation.
+     * @param[in] parentInstance In case the instance class is related to another
+     *                           addon instance class becomes with the pointer
+     *                           given to addon. Is optional and most addon types
+     *                           not use it.
+     * @return The status of addon after the creation.
+     */
+    ADDON_STATUS CreateInstance(ADDON_TYPE instanceType, const std::string& instanceID, KODI_HANDLE instance, KODI_HANDLE parentInstance = nullptr);
+
+    /*!
+     * @brief Function to destroy a on addon created instance class
+     *
+     * @param[in] instanceID The from Kodi used ID string of active instance
+     */
+    void DestroyInstance(const std::string& instanceID);
+
   protected:
     bool Initialized() { return m_initialized; }
     static uint32_t GetChildCount() { static uint32_t childCounter = 0; return childCounter++; }
```

Comment:
this isn't threadsafe, somewhat weird design to initialize on a get call and doesn't follow coding style.

Not part of this PR but yeah,